services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/SwitchOperationsService.java

```diff
@@ -252,37 +256,42 @@ public void checkSwitchHasNoIsls(SwitchId switchId) throws IllegalSwitchStateExc
      * Get switch properties.
      *
      * @param switchId target switch id
+     *
+     * @throws SwitchPropertiesNotFoundException if switch properties is not found by switch id
      */
     public SwitchPropertiesDto getSwitchProperties(SwitchId switchId) {
         Optional<SwitchProperties> result = switchPropertiesRepository.findBySwitchId(switchId);
-        return result.map(SwitchPropertiesMapper.INSTANCE::map).orElse(null);
+        return result.map(SwitchPropertiesMapper.INSTANCE::map)
+                .orElseThrow(() -> new SwitchPropertiesNotFoundException(switchId));
     }
 
     /**
      * Update switch properties.
      *
      * @param switchId target switch id
      * @param switchPropertiesDto switch properties
+     *
+     * @throws IllegalSwitchPropertiesException if switch properties are incorrect
+     * @throws SwitchPropertiesNotFoundException if switch properties is not found by switch id
      */
     public SwitchPropertiesDto updateSwitchProperties(SwitchId switchId, SwitchPropertiesDto switchPropertiesDto) {
+        if (isEmpty(switchPropertiesDto.getSupportedTransitEncapsulation())) {
+            throw new IllegalSwitchPropertiesException("Supported transit encapsulations should not be null or empty");
```

Comment:
Taking into account how you catch and handle IllegalSwitchPropertiesException and SwitchPropertiesNotFoundException, they can be checked exceptions.

Reply:
I think unchecked exception is better here. As for me checked exceptions should be used when we deal with some external resources and we need to force some cleanup. This exception may be ignored without any damage to system so I decided to make it runtime exception.